### PR TITLE
Tutorial 4: walk through updates for the deploy to GitHub Pages 

### PR DESCRIPTION
**Pull request by barriebyron**
This PR finishes the steps for tutorial 4 with the `let repoURL` updates

Preview https://docs2-git-tuto4-deploy-github-minadocs.vercel.app/zkapps/tutorials/zkapp-ui-with-react 

@ymekuria thank you for your amazing support! With your improved UI and UX and DX, we can now support different GitHub Deployment scenarios for any project

### CODE REVIEW

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx [hunk 1]
```diff
@@ -619,14 +651,26 @@ Before you can deploy your project to GitHub Pages, you must push it to a new Gi
...
-After the script builds your application, uploads it to GitHub, and GitHub processes it, your application is available at:
+1. Add two lines of code 
+1. Change to the local `/04-zkapp-browser-ui/ui/` directory.
+1. Run the deploy command:
+
+  ```sh
+  npm run deploy
+  ```
+
+The scripts defined in the `/04-zkapp-browser-ui/ui/package.json` file to build your application and upload it to GitHub.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:664]:
```suggestion
The scripts defined in the `/04-zkapp-browser-ui/ui/package.json` file do the work to build your application and upload it to GitHub. 
```

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx [hunk 2]
```diff
@@ -619,14 +670,43 @@ Before you can deploy your project to GitHub Pages, you must push it to a new Gi
 - In this tutorial, the project name is `04-zkapp-browser-ui`.
 - The `zk project` command created the correct project name strings for `04-zkapp-browser-ui` in the `next.config.js` and `src/pages/reactCOIServiceWorker.ts` files. 
 
-To deploy the UI, run the `npm run deploy` command in your local `/04-zkapp-browser-ui/ui/` directory. 
+To configure your project to deploy to GitHub Pages:
+
+1. Edit the `/04-zkapp-browser-ui/ui/ghp-postbuild.js` file.
+1. Add your repository name. For this tutorial, use `04-zkapp-browser-ui`:
```
Comment (ymekuria) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:676]: I updated the CLI to add the repository name to this file `/04-zkapp-browser-ui/ui/ghp-postbuild.js`.  We can either remove this section or mention to update the repoURL if the repo name is different from the project name when a user enters `zk project <project name>`.  In this case the project name is `04-zkapp-browser-ui` which is the same as the repo name.

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx [hunk 3]
```diff
@@ -619,14 +670,43 @@ Before you can deploy your project to GitHub Pages, you must push it to a new Gi
...
+   7 // Add your repository name here.
+   8 let repoURL = '04-zkapp-browser-ui';
+  ```
+
+1. Edit the `/zkapps/04-zkapp-browser-ui/ui/next.config.js` file.
+1. Add your repository name. For this tutorial, use `04-zkapp-browser-ui`:
+
+  ```ts ignore
+   3 // Add your repository name here.
+   4 let repoURL = '04-zkapp-browser-ui';
```
Comment (ymekuria) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:688]: If we keep this section. the code block now looks like this.

```ts
46 basePath: process.env.NODE_ENV === 'production' ? '/04-zkapp-browser-ui' : '', // update if your repo name changes for 'npm run deploy' to work successfully
47 assetPrefix: process.env.NODE_ENV === 'production' ? '/'04-zkapp-browser-ui' : '', // update if your repo name changes for 'npm run deploy' to work successfully
```

**Review by ymekuria — APPROVED** [review]
I updated the `zkApp-cli` to automate some of the manuel steps to deploy to GitHub Pages with this PR https://github.com/o1-labs/zkapp-cli/pull/468. I added some suggestions on how we can incorporate the updates here.

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx [hunk 4]
```diff
@@ -131,6 +151,23 @@ For this tutorial, run the UI commands from the local `/04-zkapp-browser-ui/ui/`
...
+✔ Would you like to use ESLint with this project? … No / Yes
+✔ Would you like to use Tailwind CSS with this project? … No / Yes
+Creating a new Next.js app in <yourfilepath>/04-zkapp-browser-ui/ui.
+
+Using npm.
+
+Initializing project with template: default
+```
+
+For this tutorial, you run commands from the root of the `04-zkapp-browser-ui` directory as you work in the `ui/src/pages` directory on files that contain the UI code.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:169]:
```suggestion
For this tutorial, you run commands from the root of the `04-zkapp-browser-ui/ui` directory as you work in the `ui/src/pages` directory on files that contain the UI code. 
```

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx (hunk 2 above)
Comment (barriebyron) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:676]:
```suggestion
```

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx [hunk 5]
```diff
@@ -619,14 +670,43 @@ Before you can deploy your project to GitHub Pages, you must push it to a new Gi
...
+1. Add your repository name. For this tutorial, use `04-zkapp-browser-ui`:
 
-After the script builds your application, uploads it to GitHub, and GitHub processes it, your application is available at:
+  ```ts ignore
+   7 // Add your repository name here.
+   8 let repoURL = '04-zkapp-browser-ui';
+  ```
+
+1. Edit the `/zkapps/04-zkapp-browser-ui/ui/next.config.js` file.
+1. Add your repository name. For this tutorial, use `04-zkapp-browser-ui`:
```
Comment (barriebyron) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:684]:
```suggestion
```

Comment (barriebyron) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:684]:
```suggestion
```

File: docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx (hunk 3 above)
Comment (barriebyron) [inline docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx:688]:
```suggestion
```

**Pull request merged**